ibtk/src/lagrangian/FEDataManager.cpp

```diff
@@ -945,9 +980,209 @@ FEDataManager::prolongData(const int f_data_idx,
+            X_node_cache.resize(n_node);
+            X_min = Point::Constant(std::numeric_limits<double>::max());
+            X_max = Point::Constant(-std::numeric_limits<double>::max());
+            for (unsigned int k = 0; k < n_node; ++k)
+            {
+                s_node_cache[k] = elem->point(k);
+                libMesh::Point& X = X_node_cache[k];
+                for (int d = 0; d < NDIM; ++d)
+                {
+                    X(d) = X_vec(X_dof_indices[d][k]);
+                    X_min[d] = std::min(X_min[d], X(d));
+                    X_max[d] = std::max(X_max[d], X(d));
+                }
+                elem->point(k) = X;
+            }
+            Box<NDIM> box(IndexUtilities::getCellIndex(&X_min[0], grid_geom, ratio),
+                          IndexUtilities::getCellIndex(&X_max[0], grid_geom, ratio));
+            box.grow(IntVector<NDIM>(1));
+            box = box * patch_box;
+
+            // Loop over coordinate directions and look for Eulerian grid points
+            // that are covered by the element.
+            intersection_ref_coords.clear();
+            intersection_indices.clear();
+
+            // Loop over the relevant range of indices.
+            for (CellIterator<NDIM> b(box); b; b++)
+            {
+                const CellIndex<NDIM>& i_c = b();
+                if (num_intersections(i_c) == 0 && patch_box.contains(i_c))
+                {
+                    libMesh::Point p;
+                    for (unsigned int d = 0; d < NDIM; ++d)
+                    {
+                        p(d) = patch_x_lower[d] + patch_dx[d] * (static_cast<double>(i_c(d) - patch_lower[d]) + 0.5);
+                    }
+                    static const double TOL = sqrt(std::numeric_limits<double>::epsilon());
+                    const libMesh::Point ref_coords = FEInterface::inverse_map(dim, X_fe_type, elem, p, TOL, false);
+                    if (FEInterface::on_reference_element(ref_coords, elem->type(), TOL))
+                    {
+                        intersection_ref_coords.push_back(ref_coords);
+                        intersection_indices.push_back(i_c);
+                        num_intersections(i_c) += 1;
+                    }
+                }
+            }
+
+            // Restore the nodal coordinates.
+            for (unsigned int k = 0; k < n_node; ++k)
+            {
+                elem->point(k) = s_node_cache[k];
+            }
+
+            // If there are no intersection points, then continue on to the next
+            // element.
+            if (intersection_ref_coords.empty()) continue;
+
+            // Evaluate the Lagrangian quantity at the Eulerian grid point and
+            // update the data on the grid.
+            for (unsigned int i = 0; i < n_vars; ++i)
+            {
+                F_dof_map_cache.dof_indices(elem, F_dof_indices[i], i);
+            }
+            get_values_for_interpolation(F_node, *F_petsc_vec, F_local_soln, F_dof_indices);
```

Comment:
@boyceg I'm having an issue with this part of the code and multiple processors.  I get an error thrown from the libMesh::PetscVector function map_global_to_local_index 

No index 192 in ghosted vector.
Vector contains [0,150)
And ghost array {276,277,278,164,279,165,282,283,288,150,289,151,290,291,292,293,156,157}

anything glaringly wrong?

Reply:
i.e. within get_values_for_interpolation.

Reply:
Were you able to resolve this? What example triggers the problem?

Reply:
I'm trying to run things again, now in parallel to get some faster simulations, and I am still getting this issue.

Reply:
if you can, please try running the application my-ex7 in our paper repo in parallel, with eliminate pressure jumps set to true.  I also have a branch off of this repo where I am playing with constructing ghosted vectors in the main routine.

Reply:
@boyceg re the prolongData functions, I am not sure if I ever fixed this parallel bug.  I ran everything in serial for the paper.

Reply:
OK, well, let's try to get it fixed. :-)

Reply:
I think that the code for getting DOF indices for non-`LAGRANGE` elements may have been broken in parallel. I think that this is fixed in `master`. Are you able to rebase off the current IBAMR master and see if this is fixed?

Reply:
yeah I can try that.  @drwells, how do I rebase?